### PR TITLE
Fix UDP timeout during DNS querying

### PR DESCRIPTION
**Pull request by irog-c**
Fix not receiving data from UDP socket by adding an asynchronous timeout.
Will add mocks in separate PR.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -40,7 +40,6 @@ IncludeCategories:
 IncludeIsMainRegex: '([-_](test|unittest))?$'
 IndentCaseLabels: true
 IndentWidth: 4
-InsertNewlineAtEOF: true
```
Comment (KyryloKupin) [inline .clang-format:43]: Why not?

File: include/resolver.hpp [hunk 2]
```diff
@@ -23,19 +26,37 @@ namespace kyrylokupin::asio::dns {
...
+            auto input = std::array<char, input_buffer_size>{};
+            auto timer = asio::steady_timer{co_await asio::this_coro::executor};
+            boost::system::error_code receive_ec;
+            auto current_retry_count = 0;
+            do {
+                co_await socket_.async_send(buffer(buf.data(), buf.size()), asio::use_awaitable);
+                ++current_retry_count;
+                timer.expires_after(std::chrono::seconds(timeout_seconds));
+                std::tie(std::ignore, receive_ec, std::ignore, std::ignore) =
+                        co_await asio::experimental::make_parallel_group(
```
Comment (KyryloKupin) [inline include/resolver.hpp:46]: There is a nicer way to make a parallel group using overloaded operators

File: .clang-format (hunk 1 above)
Comment (irog-c) [reply .clang-format:43]: Was having errors when I tried to run `clang-format` manually, when I removed this line it worked.

File: include/resolver.hpp (hunk 2 above)
Comment (irog-c) [reply include/resolver.hpp:46]: Could you tell me how exactly? I couldn't find anything in the documentation. Also can you tell the overloaded operators to `wait_for_one` like I did here?

Comment (KyryloKupin) [reply include/resolver.hpp:46]: There is an example in the [video](https://youtu.be/icgnqFM-aY4)

File: .clang-format (hunk 1 above)
Comment (KyryloKupin) [reply .clang-format:43]: What was the error?

**Pull request merged**